app/app/users/[user_id]/page.tsx

```diff
@@ -0,0 +1,96 @@
+"use client";
+import { useEffect, useState } from "react";
+import Card from "react-bootstrap/Card";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+import Spinner from "react-bootstrap/Spinner";
+import Table from "react-bootstrap/Table";
+import AppBreadCrumb from "@/components/AppBreadCrumb";
+import { getUser, useToken } from "@/utils/users";
+import { User } from "@/types/users";
+import { formatDateTime } from "@/utils/formatters";
+
+type UserColumn = {
+  name: keyof User;
+  label: string;
+  renderer?: (user: User) => string;
+};
+
+// todo: this can be extended and used for the user list table column sorting, etc
+const COLUMNS: UserColumn[] = [
+  {
+    name: "app_metadata",
+    label: "Role",
+    renderer: (user) => user.app_metadata.roles[0] || "",
+  },
+  { name: "name", label: "Name" },
+  { name: "email", label: "Email" },
+  { name: "last_ip", label: "Last IP address" },
+  { name: "logins_count", label: "Login count" },
+  {
+    name: "created_at",
+    label: "Created at",
+    renderer: (user) => formatDateTime(user.created_at),
+  },
+  {
+    name: "last_login",
+    label: "Last login",
+    renderer: (user) => formatDateTime(user.last_login),
+  },
+  {
+    name: "updated_at",
+    label: "Updated at",
+    renderer: (user) => formatDateTime(user.updated_at),
+  },
+];
+
+export default function UserDetails({
+  params,
+}: {
+  params: { user_id: string };
+}) {
+  const token = useToken();
+  const [user, setUser] = useState<User | null>(null);
```

Comment:
with not much more effort, we could pass the user data into this page component from the `users` list. the `next/router` component supports passing state data. but this is such a rarely used corner of the app that i didn't bother.